checks.d/etcd.py

```diff
@@ -133,18 +144,21 @@ def check(self, instance):
             self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.OK,
                                tags=["url:{0}".format(url)])
 
-    def _get_self_metrics(self, url, timeout):
-        return self._get_json(url + "/v2/stats/self", timeout)
+    def _get_self_metrics(self, url, ssl_params, timeout):
+        return self._get_json(url + "/v2/stats/self",  ssl_params, timeout)
 
-    def _get_store_metrics(self, url, timeout):
-        return self._get_json(url + "/v2/stats/store", timeout)
+    def _get_store_metrics(self, url, ssl_params, timeout):
+        return self._get_json(url + "/v2/stats/store",  ssl_params, timeout)
 
-    def _get_leader_metrics(self, url, timeout):
-        return self._get_json(url + "/v2/stats/leader", timeout)
+    def _get_leader_metrics(self, url, ssl_params, timeout):
+        return self._get_json(url + "/v2/stats/leader", ssl_params, timeout)
 
-    def _get_json(self, url, timeout):
+    def _get_json(self, url, ssl_params, timeout):
         try:
-            r = requests.get(url, timeout=timeout, headers=headers(self.agentConfig))
+            if ssl_params['ssl']:
+                r = requests.get(url, verify=ssl_params['ssl_cert_reqs'], cert=(ssl_params['ssl_certfile'], ssl_params['ssl_keyfile']), timeout=timeout, headers=headers(self.agentConfig))
```

Comment:
If `ssl_certfile` was `None` before you removed it from the `ssl_params` dict so here you'd likely get a `KeyError`